tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/json/FilterSerializer.scala

```diff
@@ -36,6 +35,13 @@ case class FilterSerializer @Inject()(mailboxIdFactory: MailboxId.Factory) {
   implicit val actionReads: Reads[Action] = Json.reads[Action]
   implicit val ruleWithIdReads: Reads[RuleWithId] = Json.reads[RuleWithId]
   implicit val updateReads: Reads[Update] = Json.valueReads[Update]
+  implicit val filterStateReads: Reads[FilterState] = {
+    case JsString(string) => FilterState.parse(string).fold(
+      error => JsError(error.getMessage),
+      filterState => JsSuccess(filterState)
+    )
```

Comment:
I personally do not like lonely `)` brackets alone on their lines.... Can you move that one up?